templates/api_server_deployment.yml

```diff
@@ -76,4 +82,12 @@ spec:
       - name: eirini-certs
         secret:
           secretName: #@ data.values.eirini.serverCerts.secretName
+      #@ if/end data.values.metric_proxy.cert.secret_name:
+      - name: metric-proxy-certs
```

Comment:
These aren't the only certs Cloud Controller is taking in as configuration and I understand that it's [hard-coded to only accept mTLS](https://github.com/cloudfoundry/cloud_controller_ng/blob/2ccd352b124be5c65e6e21ba9649f7691132ad6e/lib/logcache/client.rb#L9-L20) right now. Makes sense! 👍 

That said @cloudfoundry/cf-capi, is there an Issue or story anywhere around allowing these clients to make plaintext connections and relying on the Istio mesh mTLS (plus [Authorization Policies](https://istio.io/docs/reference/config/security/authorization-policy/) to secure the traffic?

Otherwise, I think the cert issuance/rotation situation needs to be thought through for these.

Reply:
I don't think we have anything in our backlog to allow plaintext connections for this client (to support Istio performing the mTLS for us)

@tcdowney What specifically do you think remains unaddressed re: cert rotation here? CCNG's code for creating this client appears to be fairly similar to other clients that we probably would like certs rotated for (e.g. Diego), so are you thinking at the Kubernetes-deployment level?

Reply:
Yes, my concerns are specific to Kubernetes. On CF4VMs we had the luxury of Credhub helping manage the provisioning/rotation of these certs. In CF4k8s we're hoping to rely on the mesh for internal certificates. We're not leveraging it here, so just looking for either stories to do so in the future or acknowledgement that y'all are thinking about this. 

cc @ndhanushkodi who might be able to link some CredHub related docs around this.

Reply:
Can we move this discussion to a Github Issue as it relates to a larger scope?

Reply:
👍 

Yep, it's a bigger issue. I'll talk to @ankeesler about coordinating on getting a `cf-for-k8s` issue created.

Reply:
https://cloudfoundry.slack.com/archives/CH9LF6V1P/p1583945872321600